orchagent/intfsorch.cpp

```diff
@@ -1375,11 +1380,22 @@ void IntfsOrch::voqSyncAddIntf(string &alias)
     Port port;
     if(gPortsOrch->getPort(alias, port))
     {
-        if(port.m_system_port_info.type == SAI_SYSTEM_PORT_TYPE_REMOTE)
+        if (port.m_type == Port::LAG)
+        {
+            if (port.m_system_lag_info.switch_id != gVoqMySwitchId)
+            {
+                return;
+            }
+            alias = port.m_system_lag_info.alias;
+        }
+        else
         {
-            return;
+            if(port.m_system_port_info.type == SAI_SYSTEM_PORT_TYPE_REMOTE)
+            {
+                return;
+            }
+            alias = port.m_system_port_info.alias;
```

Comment:
This piece of code is duplicated in another function like voqSyncDelIntf too . Can we wrap this into a helper method? This can avoid duplicating code and also make it easy to extend it in the future.

Reply:
This is done intentionally. This repeating will be there in other places also. The intention of this logic is to get whether the port or the lag is remote or not. Since we need system alias which in the Port structure,, we get the Port structure first. Since the port structure is already available and it also has the type info, I check the type in the port structure itself instead of calling another API.